hammerblade/torch/kernel/hb_tiled_for.hpp

```diff
@@ -128,13 +128,13 @@ __attribute__((noinline)) void hb_tiled_foreach_impl(
 }
 
 // Unary
-template<typename scalar_t, typename F, typename... P>
+template<typename scalar1_t, typename scalar2_t, typename F, typename... P>
```

Comment:
I am a little bit worried about this part, since this function is called by every other element-wise kernels. Can you also check whether it still works for the other element-wise kernels ?